coalib/bearlib/aspects/collections.py

```diff
@@ -9,10 +8,7 @@ class aspectlist(list):
 
     def __init__(self, seq=()):
         for item in seq:
-            if not isinstance(item, (aspectclass, aspectbase)):
-                raise TypeError(
-                    '{} is not an aspectclass or an instance of an '
-                    'aspectclass'.format(repr(item)))
+            assert_aspect(item)
         list.__init__(self, seq)
```

Comment:
`assert_aspect` could also `return` the `item`, making it possible to just do `super().__init__(map(assert_aspect, seq))`